lib/src/base_request.dart

```diff
@@ -117,11 +118,18 @@ abstract class BaseRequest {
   /// the request is complete. If you're planning on making multiple requests to
   /// the same server, you should use a single [Client] for all of those
   /// requests.
-  Future<StreamedResponse> send() async {
+  ///
+  /// If [contentTimeout] is not null the request will be aborted if it takes
+  /// longer than the given duration to receive the entire response. If the
+  /// timeout occurs before any reply is received from the server the returned
+  /// future will as an error with a [TimeoutException]. If the timout occurs
+  /// after the reply has been started but before the entire body has been read
```

Comment:
Comma before "but"? (And then probably also after "read").

Maybe "has been started" -> "has started".

Reply:
Don't think so, the '... but ...' is a single description of a condition, whereas a comma would suggest the two parts being separate sentences.

Reply:
I read "but before the entire body has been read" as parenthetical.
If it's not a parenthetical clause, it deserves to be equally prominent to the main clause, and I'd change `but` to `and`.
I'd still have a comma after `read` in either case.